### PR TITLE
resolve from references when creating builds

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@soltysh ptal.

**Comment by bparees** [comment]
[test]

File: pkg/build/generator/generator.go [hunk 1]
```diff
@@ -170,38 +174,17 @@ func (g *BuildGenerator) generateBuildUsingImageTriggerTag(ctx kapi.Context, con
...
-			return nil, err
-		}
-
-		if imageRepo == nil || len(imageRepo.Status.DockerImageRepository) == 0 {
-			continue
+		imageRef, err := g.resolveImageRepoReference(ctx, &icTrigger.From, icTrigger.Tag, config.Namespace)
+		if err != nil {
+			if _, ok := err.(fatalError); ok {
+				return nil, err
+			} else {
```
Comment (smarterclayton) [inline pkg/build/generator/generator.go]: Don't else when you return

File: pkg/build/generator/generator.go [hunk 2]
```diff
@@ -252,9 +236,50 @@ func (g *BuildGenerator) generateBuildFromConfig(ctx kapi.Context, bc *buildapi.
...
+	var namespace string
+	if len(from.Namespace) != 0 {
+		namespace = from.Namespace
+	} else {
+		namespace = defaultNamespace
+	}
+	imageRepo, err := g.Client.GetImageRepository(kapi.WithNamespace(ctx, namespace), from.Name)
+	if err != nil && errors.IsNotFound(err) {
+		return "", err
+	} else if err != nil {
```
Comment (smarterclayton) [inline pkg/build/generator/generator.go]: Don't else when you return.  Nest the ifs, or use a switch.

**Comment by bparees** [comment]
@smarterclayton ifs restructured.

**Comment by bparees** [comment]
@smarterclayton case fixed, default tag logic removed.

**Comment by soltysh** [comment]
LGTM

**Comment by bparees** [comment]
[merge]

**Comment by bparees** [comment]
[merge]

**Comment by bparees** [comment]
[merge]

**Comment by bparees** [comment]
[merge]

**Comment by bparees** [comment]
[merge]

**Pull request merged**